internal/server/access_keys_test.go

```diff
@@ -35,10 +37,17 @@ func TestAPI_CreateAccessKey(t *testing.T) {
 		body := tc.setup(t)
 
 		// nolint:noctx
-		req := httptest.NewRequest(http.MethodPost, "/api/access-keys", jsonBody(t, body))
+		req := httptest.NewRequest(http.MethodPost, "/api/access-keys", body)
 		req.Header.Set("Authorization", "Bearer "+adminAccessKey(srv))
 		req.Header.Set("Infra-Version", apiVersionLatest)
 
+		for k := range tc.headers {
+			req.Header.Del(k)
```

Comment:
If we're always calling `req.Header.Set` below we can skip this Del call since Set should always [replace](https://pkg.go.dev/net/http#Header.Set) any values for that key

Reply:
You're right. I actually thought I used `Header.Add()` which will require the `Header.Del` to ensure a proper override.